components/tor/tor_utils.cc

```diff
@@ -6,15 +6,118 @@
 #include "brave/components/tor/tor_utils.h"
 
 #include <string>
+#include <utility>
+#include <vector>
 
 #include "base/files/file_path.h"
+#include "base/no_destructor.h"
+#include "base/values.h"
 #include "brave/components/tor/tor_constants.h"
 #include "chrome/common/chrome_constants.h"
 #include "chrome/common/pref_names.h"
 #include "components/prefs/pref_service.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace {
+constexpr const char kUseBridgesKey[] = "use_bridges";
+constexpr const char kUseBuiltinBridgesKey[] = "use_builtin_bridges";
+constexpr const char kBridgesKey[] = "bridges";
+}  // namespace
 
 namespace tor {
 
+BridgesConfig::BridgesConfig() = default;
+BridgesConfig::BridgesConfig(BridgesConfig&&) noexcept = default;
+BridgesConfig::~BridgesConfig() = default;
+
+BridgesConfig& BridgesConfig::operator=(BridgesConfig&&) noexcept = default;
+
+// clang-format off
+const std::vector<std::string>& BridgesConfig::GetBuiltinBridges() const {
+  static const base::NoDestructor<std::vector<std::string>> kNoneBridges;
+
+  static const base::NoDestructor<std::vector<std::string>> kSnowflakeBridges(
+      {"snowflake 192.0.2.3:1 2B280B23E1107BB62ABFC40DDCC8824814F80A72"});
+
+  static const base::NoDestructor<std::vector<std::string>> kObfs4Briges({
```

Comment:
where are all these bridge IPs from? not sure if it's worth having the builtins at all because wouldn't they just get blocked soon?

Reply:
This is taken from the Tor browser, but I agree that this is not a very good solution, perhaps we should get a "predefined" list of bridges from a file in the component or from some network api.

Reply:
If TBB uses these then does that mean that they're not always blocked?